include/seqan3/argument_parser/detail/format_parse.hpp

```diff
@@ -229,31 +230,22 @@ class format_parse : public format_base
             return end_of_options_it;
 
         return (std::find_if(begin_it, end_of_options_it,
-            [&] (const std::string & v)
+            [&] (std::string const & current_arg)
             {
                 std::string full_id = prepend_dash(id);
 
                 if constexpr (std::same_as<id_type, char>) // short id
                 {
-                    // check if prefix of v is the same since "-oValue" "-o=Value" "-o value" are all valid
-                    return v.substr(0, full_id.size()) == full_id;
+                    // check if current_arg starts with "-o", i.e. it correctly identifies all short notations:
+                    // "-ovalue", "-o=value", and "-o value".
+                    return current_arg.substr(0, full_id.size()) == full_id;
                 }
                 else
                 {
                     // only "--opt Value" or "--opt=Value" are valid
-                    if (v.substr(0, full_id.size()) == full_id)
-                    {
-                        if (v.size() == full_id.size())
-                            return true; // value was separated by space and is next in argv
-                        else if (v[full_id.size()] == '=')
-                            return true; // value is separated by '='
-                        else
-                            return false;
-                    }
-                    else
-                    {
-                        return false;
-                    }
+                    bool valid_long_id = current_arg.substr(0, full_id.size()) == full_id && // prefix is the same
```

Comment:
why don't you return it directly?

Reply:
Because multiline return statements are ugly? :D

Reply:
Oh my, can you please directly return it? We usually don’t create extra variables if it is not part of another composition or used multiple times. I also think it does not make the code easier to understand since the expression is the same and one is wondering why there was an extra variable. I was wondering that it was meant for something else. That’s why I asked :)